### PR TITLE
Add Sushiswap Furo to DefiLlama

### PR DESCRIPTION
**Pull request by HHK-ETH**
**NOTE**

1. The protocol is usually listed within 24 hours of merging the PR
2. Please fill the form below  **only if the PR is for listing a new protocol** else it can be ignored/replaced with reason/details about the PR
3. **For updating listing info** It is a different repo, you can find your listing in this file: https://github.com/DefiLlama/defillama-server/blob/master/defi/src/protocols/data2.ts, you can  edit it there and put up a PR
4. Do not edit/push `package-lock.json` file as part of your changes, we use lockfileVersion 2, and most use v1 and using that messes up our CI
5. No need to go to our discord and announce that you've created a PR, we monitor all PRs and will review it asap

---
##### Name (to be shown on DefiLlama):
Sushiswap Furo

##### Twitter Link:
https://twitter.com/SushiSwap

##### Website Link:
https://www.sushi.com/furo

##### Logo (High resolution, preferably in .svg and .png, for application on both white and black backgrounds. Will be shown with rounded borders):
android-chrome-512x512

##### Current TVL:
170m$ (day of the pr)

##### Chain:
"ethereum", "polygon", "fantom", "bsc", "avax", "arbitrum", "optimism", "xdai", "harmony", "moonbeam", "moonriver".

##### Coingecko ID (so your TVL can appear on Coingecko): (https://api.coingecko.com/api/v3/coins/list)
sushi

##### Coinmarketcap ID (so your TVL can appear on Coinmarketcap): (https://api.coinmarketcap.com/data-api/v3/map/all?listing_status=active,inactive,untracked&start=1&limit=10000)
sushiswap

##### Short Description (to be shown on DefiLlama):
Earn, stream and automate your DAO salaries and token vesting with Furo.

##### Category (full list at https://defillama.com/categories) *Please choose only one:
Payments

##### methodology (what is being counted as tvl, how is tvl being calculated):
TVL of Furo consists of tokens deployed into a Stream or a Vesting sitting in BentoBox.

### CODE REVIEW

**Comment by llamatester** [comment]
The adapter at projects/sushiswap-furo exports TVL: 
        
 
 
    moonriver                 0
    harmony                   0
    bsc                       3.6663897073386176
    arbitrum                  0.04507254310960053
    xdai                      8.966476054873462
    avax                      28.726938171643177
    polygon                   33.06 k
    fantom                    1.992955582369
    optimism                  0.07353269462646529
    moonbeam                  0.452485
    ethereum                  171.25 M
    
    total                    171.28 M

**Comment by llamatester** [comment]
Error while running adapter at projects/sushiswap-furo: 
        
 
  
    
    
    Error: call revert exception [ See: https://links.ethers.org/v5-errors-CALL_EXCEPTION ] (method="toAmount(address,uint256,bool)", data="0x", errorArgs=null, errorName=null, errorSignature=null, reason=null, code=CALL_EXCEPTION, version=abi/5.6.4)
    Truncated error stack:
        at Interface.decodeFunctionResult (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/node_modules/@ethersproject/abi/lib/interface.js:388:23)

**Comment by HHK-ETH** [comment]
Think there might be an issue with llamatester, the adapter seems to be working properly in local.
Had the same issue with Kashi PR last week.

**Comment by llamatester** [comment]
The adapter at projects/sushiswap-furo exports TVL: 
        
 
 
    moonriver                 0
    optimism                  0
    harmony                   0
    xdai                      8.984673490784672
    arbitrum                  0.000031954200404892835
    bsc                       3.6966797049207867
    ethereum                  3.31 M
    polygon                   33.15 k
    fantom                    1.9990953122000001
    moonbeam                  0.454259
    avax                      28.886809440755986
    
    total                    3.34 M

**Comment by g1nt0ki** [comment]
@HHK-ETH thanks for the adapter. I have made a minor refactor to it.

Not a clean solution but I have added a whitelist of tokens to be counted, we do not count vested tokens in other streaming/locking protocols (llamapay, parallel, pinksale, team finance, etc) 

The reason for this, vested tokens are not yet in circulation and can be used to inflate tvl.

If you are fine with the change, I can merge this and list it.

**Comment by g1nt0ki** [comment]
> Think there might be an issue with llamatester, the adapter seems to be working properly in local. Had the same issue with Kashi PR last week.

think this is caused by xdai node being funky and not returning block height

**Comment by matthewlilley** [comment]
> @HHK-ETH thanks for the adapter. I have made a minor refactor to it.
> 
> Not a clean solution but I have added a whitelist of tokens to be counted, we do not count vested tokens in other streaming/locking protocols (llamapay, parallel, pinksale, team finance, etc)
> 
> The reason for this, vested tokens are not yet in circulation and can be used to inflate tvl.
> 
> If you are fine with the change, I can merge this and list it.

You are counting all tokens on Sablier, Super Fluid, etc... If you decide that you aren't going to count the true TVL, you should do this across all protocols, and not just Sushi.

**Comment by llamatester** [comment]
The adapter at projects/sushiswap-furo exports TVL: 
        
 
 
    moonriver                 0
    optimism                  0
    harmony                   0
    fantom                    1.9832229418360001
    avax                      28.905906467372553
    bsc                       3.743739701164327
    ethereum                  3.25 M
    xdai                      8.896323229488754
    polygon                   32.89 k
    arbitrum                  0.000031736489420854364
    moonbeam                  0.442541
    
    total                    3.29 M

**Comment by llamatester** [comment]
The adapter at projects/sablier exports TVL: 
        
 
 
    ethereum                  9.26 M
    
    total                    9.26 M

**Comment by llamatester** [comment]
The adapter at projects/superfluid.js exports TVL: 
        
 
 
    xdai                      816.4272543519269
    polygon                   894.46 k
    
    total                    895.27 k

**Comment by llamatester** [comment]
The adapter at projects/sushiswap-furo exports TVL: 
        
 
 
    moonriver                 0
    optimism                  0
    arbitrum                  0.00003197182551847771
    harmony                   0
    xdai                      8.953615453529684
    polygon                   33.17 k
    fantom                    2.0002955482
    avax                      29.15536151173349
    bsc                       3.7295697022954157
    moonbeam                  0.453242
    ethereum                  3.28 M
    
    total                    3.31 M

File: projects/sablier/index.js [hunk 1]
```diff
@@ -1,61 +1,58 @@
...
+  "ETHA": "0x59e9261255644c411afdd00bd89162d09d862e38",
+  "cDAI": "0x5d3a536e4d6dbd6114cc1ead35777bab948e3643",
+  "DAI": "0x89d24a6b4ccb1b6faa2625fe562bdd9a23260359",
+  "MATIC": "0x7d1afa7b718fb893db30a3abc0cfc608aacfebb0",
+  "AAVE": "0x7fc66500c84a76ad7e9c93437bfc5ac33e2ddae9",
+  "USDC": "0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48",
+  "UST": "0xa47c8bf37f92abed4a126bda807a7b7498661acd",
+  "stETH": "0xae7ab96520de3a18e5e111b5eaab095312d7fe84",
+  "WETH": "0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2",
+  "COMP": "0xc00e94cb662c3520282e6f5717214004a7f26888",
```
Comment (HHK-ETH) [inline projects/sablier/index.js:36]: Hey @g1nt0ki, considering blueships are not taken into account for Superfluid and Furo should'nt we remove AAVE, stkAAVE, COMP and LINK from the whitelisted tokens for Sablier ?

File: projects/sushiswap-furo/furo.js [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+  optimism: "0xc35DADB65012eC5796536bD9864eD8773aBc74C4",
+  xdai: "0xE2d7F5dd869Fc7c126D21b13a9080e75a4bDb324",
+  harmony: "0xA28cfF72b04f83A7E3f912e6ad34d5537708a2C2",
+  moonbeam: "0x80C7DD17B01855a6D2347444a0FCC36136a314de",
+  moonriver: "0x145d82bCa93cCa2AE057D1c6f26245d1b9522E6F",
+};
+
+const furoQuery = gql`
+  query get_tokens($block: Int) {
+    tokens(
```
Comment (HHK-ETH) [inline projects/sushiswap-furo/furo.js:43]: Also I forgot to add `block: { number: $block } ` to the query, might want to add it to get historical TVL.

**Comment by llamatester** [comment]
The adapter at projects/sablier exports TVL: 
        
 
 
    ethereum                  9.33 M
    
    total                    9.33 M

**Comment by llamatester** [comment]
The adapter at projects/superfluid.js exports TVL: 
        
 
 
    xdai                      814.616004990596
    polygon                   899.26 k
    
    total                    900.08 k

**Comment by llamatester** [comment]
The adapter at projects/sushiswap-furo exports TVL: 
        
 
 
    optimism                  0
    moonriver                 0
    harmony                   0
    xdai                      9.04471962945905
    polygon                   33.37 k
    avax                      29.16066316361318
    ethereum                  3.30 M
    fantom                    2.0122955981
    arbitrum                  0.00003213046832204085
    moonbeam                  0.45122
    bsc                       3.7373697016727983
    
    total                    3.33 M

**Comment by llamatester** [comment]
The adapter at projects/llamapay exports TVL: 
        
 
 
    xdai-vested               0
    arbitrum                  8.77 k
    arbitrum-vested           192.72 k
    xdai                      43.312036
    avax                      293.23 k
    avax-vested               16.35 k
    optimism                  43.75 k
    polygon                   163.95 k
    optimism-vested           1.7953857054067504
    ethereum                  2.05 M
    fantom                    108.06 k
    polygon-vested            2.00 k
    fantom-vested             5.75 k
    bsc                       72.83 k
    metis-vested              0
    metis                     11.034240518777777
    ethereum-vested           1.82 M
    meter                     220
    bsc-vested                254.72 k
    meter-vested              84.64 k
    vested                    2.37 M
    
    total                    2.74 M

**Comment by llamatester** [comment]
The adapter at projects/sablier exports TVL: 
        
 
 
    ethereum                  7.28 M
    ethereum-vested           227.91 M
    vested                    227.91 M
    
    total                    7.28 M

**Comment by llamatester** [comment]
The adapter at projects/superfluid.js exports TVL: 
        
 
 
    xdai                      776.3282118404157
    xdai-vested               29.07 k
    polygon                   906.75 k
    polygon-vested            5.17 M
    vested                    5.20 M
    
    total                    907.52 k

**Comment by llamatester** [comment]
The adapter at projects/sushiswap-furo exports TVL: 
        
 
 
    moonriver-vested          0
    moonriver                 0
    xdai-vested               0
    bsc-vested                0
    optimism                  0
    fantom-vested             0
    moonbeam-vested           0
    xdai                      9.008720745361702
    arbitrum-vested           0.67831532
    polygon                   31.73 k
    harmony-vested            0
    harmony                   0
    optimism-vested           0.06917726322785724
    arbitrum                  0.00003200144554478768
    avax                      10.010429303016565
    avax-vested               0.013033038999999998
    moonbeam                  0.44484
    bsc                       3.706169704163268
    fantom                    3.9847243513
    polygon-vested            0
    ethereum-vested           109.32 M
    ethereum                  3.27 M
    vested                    109.32 M
    
    total                    3.30 M

**Pull request merged**